### PR TITLE
Fixes https://github.com/umccr/RNAsum/issues/75

### PR DESCRIPTION
**Pull request by skanwal**
Update check to read in kallisto input if provided.

While testing against test data in `/data/test_data/final/test_sample_WTS`, I was getting error for sunburst_plot:

```
Error in if (!is.na(sunburst_plot[[2]])) { : the condition has length > 1
```
To resolve the issue, I have updated check to look for specific element in sublist. Think this was a warning in previous versions of R but now produces an error.

At [line 1304](https://github.com/umccr/RNAsum/compare/fix_issue_75?expand=1#diff-10a626899fa0f1760d680ae0c2e42a9651988dabeb89a11185f0f7f0f3a57491R1304), I was hitting:

```
R argument is of length zero
``` 

Think this was the issue here https://stackoverflow.com/questions/27350636/argument-is-of-length-zero-in-if-statement.
Using `isTRUE` resolves it.

### CODE REVIEW

**Review by pdiakumis — COMMENTED** [review]
Thanks Sehrish!

- No idea if the `==` to `%in%` change makes sense, just keep in mind they are quite different functions.

- Can you double check the `["x"]` subsetting you're doing? I can see different calls:

```
!is.na(sunburst_plot[[2]]["x"])
!is.na(sunburst_plot[[2]])["x"]
```

**Comment by skanwal** [comment]
Thanks for the review @pdiakumis.

> No idea if the == to %in% change makes sense, just keep in mind they are quite different functions.

Yep, understand this. Logically, %in% makes sense to me as we are trying to check if either fusion geneA or geneB are present in known_translocations.  Not sure if it makes sense to use different logic for both fusion genes. I have checked aginst the test data and FusionGDB links to both geneA and geneB work.

Context, with ==, we get following error:

```
label: arriba_filtering (with options) 
List of 4
 $ comment: logi NA
 $ message: logi FALSE
 $ warning: logi FALSE
 $ eval   : logi TRUE

[1] "Start: 2022-11-24 15:50:00"

Quitting from lines 3395-3508 (RNAseq_report.Rmd) 
Error in if (geneB == known_translocations$effector_gene) { : 
  the condition has length > 1
Calls: <Anonymous> ... withVisible -> eval_with_user_handlers -> eval -> eval
In addition: There were 50 or more warnings (use warnings() to see the first 50)
Execution halted
```
> Can you double check the ["x"] subsetting you're doing? I can see different calls.

Thanks for catching this, [fixed](https://github.com/umccr/RNAsum/commit/a398fa374bdaa435e7749ef49f53342b61f0b86f) and retested as well.

**Comment by pdiakumis** [comment]
LGTM!

File: rmd_files/RNAseq_report.Rmd [hunk 1]
```diff
@@ -4398,7 +4400,7 @@ rm(counts_vs_transformed, data_transformation_nonfiltered, data_transformation_f
 
 Table summarising **all altered genes** listed across following report sections: `r if ( runPcgrChunk ) { c("[Mutated genes], ") }` `r if ( runFusionChunk ) { c("[Fusion genes] (supported by genomic data or reported in FusionGDB), ") }` `r if ( runSVsChunk ) { c("[Structural variants], ") }` `r if ( runPurpleChunk ) { paste0("[CN altered genes] (CN values =< ", cn_bottom, " or >= ", cn_top, " and reported as cancer genes)") }` [Immune markers] or [HRD genes]. The *Resources* column contains links to databases that may provide additional source of evidence for the altered genes' clinical significance. Genes ordered by the number of report sections they appear in (*Count* column) and then alphabetically.
 
-`r if ( is.na(sunburst_plot[[2]]) ) { c("<span style=\"color:#ff0000\">NOTE</span>, no genes are listed in more then one section.") }`
+`r if ( is.na(sunburst_plot[[2]])["x"] ) { c("<span style=\"color:#ff0000\">NOTE</span>, no genes are listed in more then one section.") }`
```
Comment (pdiakumis) [inline rmd_files/RNAseq_report.Rmd:4403]: @skanwal and here.

Comment (skanwal) [reply rmd_files/RNAseq_report.Rmd:4403]: updated in https://github.com/umccr/RNAsum/pull/77

File: rmd_files/RNAseq_report.Rmd [hunk 2]
```diff
@@ -4354,17 +4356,17 @@ treatment_timeline
...
```
[the remark this answers is not in the document]
Comment (skanwal) [reply rmd_files/RNAseq_report.Rmd:4361]: updated in https://github.com/umccr/RNAsum/pull/77

**Pull request merged**